src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -793,9 +803,15 @@ static int client_message_init(
         if (!packet)
                 return -ENOMEM;
 
-        r = dhcp_message_init(&packet->dhcp, BOOTREQUEST, client->xid, type,
-                              client->arp_type, client->hw_addr.length, client->hw_addr.bytes,
-                              optlen, &optoffset);
+        if (client->bootp) {
```

Comment:
Please drop unnecessary `{}`. Also, please drop the empty line before the error check.
```c
if (client->bootp)
        r = bootp_message_init(...);
else
        r = dhcp_message_init(...);
if (r < 0)
        return r;
```

Reply:
One pickup I found from the unit test is that `optoffset` needs to be initialized to 0 in the `bootp_message_init()` path. This is added to the upcoming push.